### PR TITLE
Build pipeline improvements

### PR DESCRIPTION
**Pull request by smoralesd**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Added relevant unit test for your changes. (`npm run test`)
- [ ] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] Ran precheckin (`npm run precheckin`)
- [ ] Added screenshots/GIFs for UI changes.

#### Description of changes

Use templates to reduce duplication on our build pipeline definition.

#### Notes for reviewers

The end goal should be adding a ubuntu job. I have found issues running unit and e2e test so we need more time to fix those issues. This is preparation work and anything related to running our build under ubuntu is out of scope.

### CODE REVIEW

File: pipeline/e2e-test.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+steps:
+    - task: Npm@1
+      inputs:
+          command: custom
+          customCommand: run build:all
+      displayName: npm run build:all
+
+    # e2e tests depend on npm run drop
```
Comment (peterdur) [inline pipeline/e2e-test.yaml]: Do the e2e tests really depend on `npm run drop`? Because we don't seem to be doing that.

Comment (smoralesd) [reply pipeline/e2e-test.yaml]: Comment is misleading. The actual dependency is on the `drop` folder, not on `npm run drop`. So, we need to run some of our steps to generate that `drop` folder and I guess the simplest, most straightforward way to do this is by running `npm run build:all`.

I updated the comment to reflect this.

File: pipeline/e2e-test.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+steps:
+    # e2e tests depend drop folder. run build:all will create said folder
```
Comment (dbjorge) [inline pipeline/e2e-test.yaml]: I think including the 2 "publish" tasks for the e2e results/screenshots is at least as reasonable as including build:all here :)

Comment (smoralesd) [reply pipeline/e2e-test.yaml]: I though about that... in the long run though, if we are able to run unit/e2e test on ubuntu, would we want to publish test results from that run too??? So I didn't move it, it should be a problem we solve in the future.

Comment (dbjorge) [reply pipeline/e2e-test.yaml]: I think we definitely would want to publish test results from both places; we'd need to in order to debug failures effectively.

Comment (smoralesd) [reply pipeline/e2e-test.yaml]: I prefer we make those changes when we get there, basically YAGNI

**Pull request merged**